ops/testing.py

```diff
@@ -1278,6 +1293,14 @@ def _storage_attach(self, storage_id: str):
         # re-attachment of a storage unit.  This is not present in
         # ops.model._ModelBackend.
         name, index = storage_id.split('/', 1)
```

Comment:
again the name `storage_id` is no longer accurate, we should be uniform in how we call that thing, whatever it is

Reply:
It actually is accurate.  I am trying to make things consistent for users. There is other api and docs that uses "storage_id" to refer to the fully-qualified "<name>/<index>" form.  So the most consistent/cleanest way I saw to improve things for users was to preserve this terminology while trying to make the framework internals change to use the name+index convention.  So id=whole-thing.  name=name.  index=number.  Basically I made it so that all the places (including user facing ones) that were already using storage_id to refer to the entire label (including index) didn't have to change.  The only user facing change is deprecating the super confusing, overloaded "id" attribute on the storage class.